plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -954,4 +958,13 @@ public static long getModificationTime(String path, TrinoFileSystem fileSystem)
             throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Failed to get file modification time: " + path, e);
         }
     }
+
+    public static Map<String, String> loadManifestCachingProperties(Map<String, String> properties, IcebergConfig icebergConfig)
+    {
+        properties.put(IO_MANIFEST_CACHE_ENABLED, "true");
```

Comment:
this is configuring `org.apache.iceberg.ManifestFiles#CONTENT_CACHES`, right?
the cache is keyed by `FileIO` and our `ForwardingFileIo` is typically short-lived. how effective is this caching?

cc @alexjo2144 @electrum

Reply:
Yes, in one of our production environments, a big sql which was planned for 12s has been reduced to 8s~

In our internal implementation, we also used a Caffeine cache with the key `session user` and the value `FileIo` to make a `ForwardingFileIo` live longer, i can also add relevant codes to our PR

Reply:
the 12s -> 8s improvement -- was it with or without fileio caching?

what would the numbers be if we don't do this PR and instead enable filesystem caching that we already have in trino?

Reply:
> the 12s -> 8s improvement -- was it with or without fileio caching?
> 
> what would the numbers be if we don't do this PR and instead enable filesystem caching that we already have in trino?

@findepi the 12s -> 8s improvement was with fileio/manifest caching.  

Sorry our newest production environment is 423, so i couldn't test the performance about filesystem caching, but i saw that `PrestoDB` have `manifest caching` and `filesystem caching` both, so i thought that they may not conflict with each other~ The metadata caching in memory is also faster than caching in disks in most cases.  

Thank you for your advice, I have modified some codes accordingly, please review again when you have time~

Reply:
> @findepi the 12s -> 8s improvement was with fileio/manifest caching.

thanks!

before merging this PR, we should understand what's the level of improvement that it brings (in isolation).

cc @sopel39 @raunaqmorarka